core/controller/src/main/scala/whisk/core/loadBalancer/SeedNodesProvider.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.core.loadBalancer
+
+import akka.actor.Address
+
+import scala.collection.immutable.Seq
+
+trait SeedNodesProvider {
+  def getSeedNodes(): Seq[Address]
+}
+
+class StaticSeedNodesProvider(seedNodes: String, actorSystemName: String) extends SeedNodesProvider {
```

Comment:
Can you change this signature to be `StaticSeedNodesProvider(config: WhiskConfig, actorSystem: ActorSystem)`? otherwise it will need to either be wrapped or further refactored to make it extensible in the future

Reply:
I think that passing only what is indeed _needed_ to a method is the right approach in this case, e.g. two strings instead of the overhead of the rest of the whiskConfig and the actor system. It makes it easier to test, so no extra mocking is required. If another implementation would be needed, one could create another class extending `SeedNodesProvider` with custom parameters and no interference.

Reply:
Maybe what isn't clear is that way SPIs work is that the behave as factories, where each impl uses the same interface, so this impl will have to be refactored to become extensible in any way. To prepare for that you would need to have an extra class `StaticSeedNodesProviderFactory` which knows how to construct your `StaticSeedNodesProvider` using that existing constructor - my ask is to create both classes now, so that making it extensible in the future is ONLY a matter of how `StaticSeedNodesProviderFactory` is instantiated.

Reply:
Pointing out that it is needed makes it not pre-mature, but if you are not comfortable implementing it, someone will create a PR for this in the near future.

Reply:
It is not about being not comfortable. Pre-mature in this case means that you are asking for the change that won't be used in our production env now, and probably by anyone else yet. The mesos and kube support is not officially supported yet, and I believe that an spi should come with another feature and it is not necessary for this particular PR.